### DISCUSSION TITLE
Add Ligatures to fonts without

### DISCUSSION THREAD

**Discussion by ryanoasis** [ryanoasis/nerd-fonts#739] [Ideas]
Any interest in looking into and attempting to add ligatures to fonts without?

**Comment by lilyball** [comment]
I would also be interested in having an option that adds programming ligatures, like those talked about in https://www.hanselman.com/blog/MonospacedProgrammingFontsWithLigatures.aspx.

**Comment by michaeltlombardi** [comment]
Might be able to leverage [ligaturizer](https://github.com/rojiani/Ligaturizer/) for this.

**Comment by ryanoasis** [comment]
@michaeltlombardi Thanks. That looks cool! Hmm poked around not sure about the future of project though..

**Comment by michaeltlombardi** [comment]
I was thinking of using it to have a personal hacked version of the knack font for my powerline setup. :)

**Comment by ryanoasis** [comment]
@michaeltlombardi Let us know if you do and how it turns out. I would be willing to use something like this in Nerd Fonts

**Comment by michaeltlombardi** [comment]
I'm actually currently using furacode, which already included the ligatures. Will set aside time to run it against knack and report back. :)

**Comment by hronro** [comment]
+1

**Comment by RavenHursT** [comment]
Yes please :-)

**Comment by Xinkle** [comment]
+1

**Comment by ryanoasis** [comment]
This _might_ have to a be `3.0` release. Some of the glyphs are clobbering the alphabetic presentation forms (ligatures) and would need to be moved to another range. We will only use Private Use Areas going forward to prevent such further conflicts.

https://jrgraphix.net/r/Unicode/FB00-FB4F

**Comment by Umbrien** [comment]
How about adding <3 ligature? There should be more love in our code! :heart:

**Comment by Jab2870** [comment]
If you decide to go down the ligature route, would it be possible to add bigger icons, perhaps by prepending them with a special character.

As an example, I use the patched Iosevka font, which is a thin font, and this results in a lot of the icons being very small, as shown in this screenshot. The icons are (obviously) constrained by the width of the font.

![image](https://user-images.githubusercontent.com/1909202/78544934-d0608980-77f2-11ea-938a-d41088a41389.png)

If we could have the icons span 2 characters, by using a ligature, this would in my case, make them a lot easier to distinguish.

**Comment by timtro** [comment]
> If you decide to go down the ligature route, would it be possible to add bigger icons, perhaps by prepending them with a special character.
> 
> As an example, I use the patched Iosevka font, which is a thin font, and this results in a lot of the icons being very small, as shown in this screenshot. The icons are (obviously) constrained by the width of the font.
> 
> ![image](https://user-images.githubusercontent.com/1909202/78544934-d0608980-77f2-11ea-938a-d41088a41389.png)
> 
> If we could have the icons span 2 characters, by using a ligature, this would in my case, make them a lot easier to distinguish.

From what I understand, the unicode standard specifies the number of cells for each codepoint. So the renderer will need to agree to incorrectly draw the character across more cells.

**Comment by Finii** [comment]
> If you decide to go down the ligature route, would it be possible to add bigger icons, perhaps by prepending them with a special character.

Normally if you use the "Nerd Font" and not the "Nerd Font Mono" version the icons should be bigger:
I.e. https://github.com/ryanoasis/nerd-fonts/blob/master/patched-fonts/Iosevka/Regular/complete/Iosevka%20Nerd%20Font%20Complete.ttf

![image](https://user-images.githubusercontent.com/16012374/145963794-813536fd-6cea-41c2-8a39-1b728f3c4de8.png)

**Comment by ryanoasis** [comment]
converting this to a discussion, FYI
